packages/sku/src/types/types.ts

```diff
@@ -226,6 +227,22 @@ export interface SkuConfig {
    */
   dangerouslySetWebpackConfig?: (skuWebpackConfig: any) => any;
 
+  /**
+   * This function provides a way to modify sku's Vite configuration.
+   * It should only be used in exceptional circumstances where a solution cannot be achieved by adjusting standard configuration options.
+   *
+   * Before customizing your Vite configuration, please reach out in [#sku-support](https://seek.enterprise.slack.com/archives/CDL5VP5NU) to discuss your requirements and potential alternative solutions.
+   *
+   * As sku creates two Vite configs (`client` & `server|render`), this function will actually run twice.
+   * If you only need to modify one of these configs, then you can check `config.mode` within.
+   *
+   * Sku provides no guarantees that its Vite configuration will remain compatible with any customizations made within this function.
+   * It is the responsibility of the user to ensure that their customizations are compatible with sku.
+   *
+   * @link https://seek-oss.github.io/sku/#/./docs/configuration?id=dangerouslysetviteconfig
+   */
+  dangerouslySetViteConfig?: (skuViteConfig: InlineConfig) => any;
```

Comment:
Can this use [the type from the `config` plugin hook](https://vite.dev/guide/api-plugin#config), since it's just being passed into that hook anyway? This would provide `mode` and `command`, both of which can be used to scope the config changes to certain builds.

Also, I'd prefer this to be prefixed with `unstable` or something similar until we're sure we want to stabilize it.